### PR TITLE
NXP backend: Add conversion and quantization support for dim_order_ops._clone_dim_order.default

### PR DESCRIPTION
**Pull request by StrycekSimon**
### Summary
- Adds support for conversion and quantization of `dim_order_ops._clone_dim_order.default` operator and fixes problems with some variations of `nn.Dropout`.
- Adds more robust test cases for clone operators.

### Test plan
All changes should be covered by unit tests.

cc @robert-kalmar @JakeStevens @digantdesai

### CODE REVIEW

File: backends/nxp/tests/executors.py [hunk 1]
```diff
@@ -370,7 +370,11 @@ def convert_run_compare(
 
 
 def graph_contains_any_of_ops(graph: Graph, ops: list) -> bool:
-    return any(node.target in ops for node in graph.nodes)
+    return graph_contains_any(graph, condition=lambda n: n in ops)
```
Comment (MartinPavella) [inline backends/nxp/tests/executors.py]: Shouldn't this be:
```python
return graph_contains_any(graph, condition=lambda n: hasattr(n, "target") and n.target in ops)
```
?

File: backends/nxp/tests/executors.py [hunk 2]
```diff
@@ -370,7 +370,11 @@ def convert_run_compare(
...
+
+
+def graph_contains_any(graph: Graph, condition: Callable[[Node], bool]) -> bool:
+    return any(condition(node.target) for node in graph.nodes)
```
Comment (MartinPavella) [inline backends/nxp/tests/executors.py]: This should be:
```python
return any(condition(node) for node in graph.nodes)
```

or if you prefer:
```python
return any(map(condition, graph.nodes))
```

File: backends/nxp/tests/ir/converter/node_converter/test_clone_converter.py [hunk 3]
```diff
@@ -74,57 +75,117 @@ def forward(self, x):
...
-    converter_spy = mocker.spy(EdgeProgramToIRConverter, "convert_program")
+    @staticmethod
+    def _node_is_clone(node) -> bool:
+        clone_ops = [
+            exir_ops.edge.aten.clone.default,
+            exir_ops.edge.dim_order_ops._clone_dim_order.default,
+        ]
 
-    quantized_program = to_quantized_edge_program(model, input_shape).exported_program()
+        def target_can_be_clone(node):
```
Comment (MartinPavella) [inline backends/nxp/tests/ir/converter/node_converter/test_clone_converter.py:94]: 👍🏻

File: backends/nxp/tests/ir/converter/node_converter/test_clone_converter.py [hunk 4]
```diff
@@ -74,57 +75,117 @@ def forward(self, x):
...
+    def test_conv_dropout_no_quant(
+        self, inplace_dropout: bool, input_shape: tuple[int]
+    ):
+        model = SingleConvBlockWithDropout(
+            conv_in_channels=input_shape[1], perform_inplace_dropout=inplace_dropout
+        ).eval()
+
+        edge_program = to_edge_program(model, input_shape).exported_program()
+
+        has_clone = graph_contains_any(
```
Comment (MartinPavella) [inline backends/nxp/tests/ir/converter/node_converter/test_clone_converter.py]: Nit:
You can use the `graph_contains_any_of_ops()` here (once it's fixed).

**Review by MartinPavella — CHANGES_REQUESTED** [review]

**Comment by MartinPavella** [comment]
Please take a look at the failing Neutron tests.

**Comment by StrycekSimon** [comment]
Both fails seems unrelated to this PR while one of them was already failing in our previous PRs.

File: backends/nxp/tests/ir/converter/node_converter/test_clone_converter.py [hunk 5]
```diff
@@ -74,57 +76,108 @@ def forward(self, x):
...
+            assert not graph_contains_any(
+                graph=quantized_program.graph,
+                condition=TestCloneConverter._node_is_clone,
+            )
+
+            input_data = (np.random.random(input_shape) * 50).astype(np.int8)
+            convert_run_compare(
+                exported_program,
+                tfl_model=tflite_flatbuffers_model,
+                tflite_input_preprocess=ToNHWCPreprocess(),
```
Comment (roman-janik-nxp) [inline backends/nxp/tests/ir/converter/node_converter/test_clone_converter.py]: Please change it to `ToChannelLastPreprocess()` resp. `ToChannelFirstPreprocess()` as current calls are deprecated.

File: backends/nxp/tests/ir/converter/node_converter/test_clone_converter.py [hunk 6]
```diff
@@ -74,57 +76,108 @@ def forward(self, x):
...
```
Comment (roman-janik-nxp) [inline backends/nxp/tests/ir/converter/node_converter/test_clone_converter.py]: Please change it to `ToChannelLastPreprocess()` resp. `ToChannelFirstPreprocess()` as current calls are deprecated.

File: backends/nxp/tests/ir/converter/node_converter/test_clone_converter.py [hunk 7]
```diff
@@ -74,57 +76,108 @@ def forward(self, x):
...
+                input_data=input_data,
+                atol=1.0,
+            )
+
+    @classmethod
```
Comment (roman-janik-nxp) [inline backends/nxp/tests/ir/converter/node_converter/test_clone_converter.py]: Please move this method up to be first in class.

**Review by roman-janik-nxp — CHANGES_REQUESTED** [review]
Please update and rebase on new main.

**Comment by StrycekSimon** [comment]
> Please update and rebase on new main.

That is done. ✅

**Review by roman-janik-nxp — APPROVED** [review]

**Review by MartinPavella — APPROVED** [review]
